### ISSUE TITLE
Warn user when exiting edit mode with unsaved data

### ISSUE THREAD

**Issue by erikphansen** [department-of-veterans-affairs/va.gov-team#9769]
If a user has started editing part of their profile and hit the CANCEL button to exit edit mode without saving, we should warn them that they will lose the info they have entered.

We will need to decide on a UX for this. Show a modal? Show an alert somewhere else? The only thing that we have like this, that I know of, is getting warned when you are filling out a benefit application as a logged-out user and attempt to leave the page. I believe in that scenario you are shown a modal warning you that you will lose data.

**Comment by tressaellen** [comment]
![Screen Shot 2020-06-09 at 12.25.08 PM.png](https://images.zenhubusercontent.com/5dfa65bbbd078d1699e52294/706d9646-02ee-4961-898e-2fac651d857f)

Are you sure?

You haven’t finished editing, if you cancel now then all of your progress will be lost.

[Button] Cancel
[Button] Continue editing

@peggygannon I'd love some help with this modal - especially the verbiage on the buttons.

**Comment by peggygannon** [comment]
Hi, @tressaellen Does the "Are you sure? " message come after this one below? 

**You’re editing your [section]** (no period)
Please save or cancel your work before editing a new section of your profile. If you cancel your editing, your in-progress work won’t be saved. 
Should the buttons read?
Cancel
Save

**Comment by tressaellen** [comment]
@peggygannon yes it could come after this message, it depends on if we can edit from the modal or not. If we can then if a user selects 'cancel' then it would show this message. If we can't then the user would return to the edit screen, select cancel there and then it would show this message. Does that make sense?

**Comment by tressaellen** [comment]
Erik says we cannot trigger a save form within a modal, so this would pop up if a user tried to exit edit mode without saving.  Does this work?

<img width="1144" alt="Screen Shot 2020-06-15 at 1 53 19 PM" src="https://user-images.githubusercontent.com/58752772/84700222-47a34f80-aeff-11ea-888f-2063a27d43f8.png">

Are you sure?
You haven’t finished editing your [section]. If you cancel, your in-progress work won't be saved.
[Button] Continue editing
[Button] Cancel

**Comment by peggygannon** [comment]
@tressaellen  hello, yes, this message would work. So the flow would be: 

**You’re editing your [section]** (no period)
Please go back and save or cancel your work before editing a new section of your profile. If you cancel, your in-progress work won’t be saved.
{button} OK

If user tries to leave section w/out saving first, then this message would appear:

**Are you sure?**
You haven’t finished editing your [section]. If you cancel, your in-progress work won't be saved.
[Button] Continue editing
[Button] Cancel

**Comment by tressaellen** [comment]
Oh, I just realized I dropped in the wrong screenshot in the last comment, sorry about that! But yes, that messaging would be perfect. Thank you!

Are you sure?
You haven’t finished editing your [section]. If you cancel, your in-progress work won't be saved.
[Button] Continue editing
[Button] Cancel

<img width="1082" alt="Screen Shot 2020-06-16 at 11 42 48 AM" src="https://user-images.githubusercontent.com/58752772/84808869-c5299700-afb5-11ea-8a80-6a08fc76daaa.png">
